### PR TITLE
Add trackdisambig from musicbrainz.

### LINKED ISSUE #1904: Fetch MusicBrainz track disambiguation field
I download a lot of singletons, and I use the track disambiguation notes frequently. I saw that Beets has a way to note the **album** disambig, but not the **track** disambig. Can this be inserted into a future release? This way, my beets.yaml config could look something like this:

`paths:
    default: $albumartist_sort/[$year] $album/$track. $artist- $title
    comp: Various Artists/[$year] $album/$track. $artist- $title
    singleton: $artist_sort/$artist- $title [$trackdisambig]
`

### PR DESCRIPTION
**Pull request by m0zes**
fixes beetbox/beets#1904

## Description

Fixes #1904.

(...)

## To Do

- [x] Documentation. (If you've add a new command-line flag, for example, find the appropriate page under `docs/` to describe it.)
- [x] Changelog. (Add an entry to `docs/changelog.rst` near the top of the document.)
- [x] Tests. (Encouraged but not strictly required.)

### CODE REVIEW

File: beets/autotag/hooks.py [hunk 1]
```diff
@@ -163,7 +163,7 @@ def __init__(self, title=None, track_id=None, release_track_id=None,
                  composer=None, composer_sort=None, arranger=None,
                  track_alt=None, work=None, mb_workid=None,
                  work_disambig=None, bpm=None, initial_key=None, genre=None,
-                 **kwargs):
+                 trackdisambig=None, **kwargs):
```
Comment (sampsyo) [inline beets/autotag/hooks.py:166]: This addition is actually not necessary anymore, now that we use `kwargs` as a catch-all for arbitrary field names. The list here of old field names is just for backwards-compatibility.

File: beets/autotag/mb.py [hunk 2]
```diff
@@ -223,6 +223,9 @@ def track_info(recording, index=None, medium=None, medium_index=None,
     if recording.get('length'):
         info.length = int(recording['length']) / (1000.0)
 
+    if recording.get('disambiguation'):
+        info.trackdisambig = recording.get('disambiguation')
```
Comment (sampsyo) [inline beets/autotag/mb.py:227]: This can be `recording['disambiguation']` because it's guaranteed to exist at this point, I believe.

**Review by sampsyo — COMMENTED** [review]
Yay yay; this looks fantastic! I have a couple of comments on minor details within; let me know what you think.

**Comment by m0zes** [comment]
Thanks for the review. 

I've made some changes based on your comments. Let me know what you think.

File: beets/autotag/mb.py [hunk 3]
```diff
@@ -223,6 +223,8 @@ def track_info(recording, index=None, medium=None, medium_index=None,
     if recording.get('length'):
         info.length = int(recording['length']) / (1000.0)
 
+    info.trackdisambig = recording.get('disambiguation', None)
```
Comment (sampsyo) [inline beets/autotag/mb.py:226]: This seems like a perfectly fine way to do it, but it's exactly equivalent to `recording.get('disambiguation')` (the default default value is `None`!).

File: docs/changelog.rst [hunk 4]
```diff
@@ -178,6 +178,9 @@ New features:
   :bug:`3478`
 * Removes usage of the bs1770gain replaygain backend.
   Thanks to :user:`SamuelCook`.
+* Added ``trackdisambig`` which stores the recording disambiguation from
+  musicbrainz for each track.
```
Comment (sampsyo) [inline docs/changelog.rst:182]:
```suggestion
  MusicBrainz for each track.
```

**Review by sampsyo — COMMENTED** [review]
Neato. At the risk of getting ridiculously detailed, there is one small way this could be simpler! This is essentially ready to merge, however.

**Review by sampsyo — APPROVED** [review]
Yay! Many thanks!!

**Pull request merged**